packages/iris-grid/src/IrisGridModel.ts

```diff
@@ -131,6 +132,16 @@ abstract class IrisGridModel<
     return this.columns;
   }
 
+  /** List of column movements defined by the model. Used as initial movements for IrisGrid */
+  get movedColumns(): MoveOperation[] {
```

Comment:
I'm going to need to think about this a bit... but essentially `movedColumns` doesn't belong in IrisGridModel, since it doesn't know about visible stuff.
Before, it applied all the layoutHint stuffs on the underlying model, translating the columns before upper layers consumed.

Now, I think it more appropriate if you just pass up the layout hints, and IrisGrid translates those to the appropriate `movedColumns`. Since IrisGrid and Grid concern themselves with the layout, whereas the models are just access to the data.

Reply:
Of course, upon reflection, I realize `floatingLeftCount` and all those options are dependent on the visual order... even those should just be at the Grid/IrisGrid level instead of GridModel/IrisGridModel. What's the right path going forward? Going to need to think about that one a bit.

Reply:
It might be time to start designing Grid V2 or at least documenting what goes where with reasons. I think the component/model/renderer/metrics are slowly starting to overlap somewhat and the boundary is becoming less clear.

In my opinion the actual component shouldn't do much more than pull in all the parts and tie them together w/ the JS API

Reply:
Agreed, the division is not clear. Especially with things like the floating columns and floating rows.

In terms of Grid v2 and responding to some of your other ideas:
- Treating floating grids as separate grids I think is a great idea, would just need to make sure the grids stay in sync.
- Treating events kind of like it is a DOM (such as subscribing to a `headerEvent` or something). The raw mouse event handlers already have the coordinate/cell that's clicked on, perhaps some more info could be added to that. Exposing the raw mouse events/handlers that we do right now on just the grid canvas is quite flexible, but it gets sticky whenever we mix actual DOM in though (such as for advanced filters and input tables). In reality I think we should simplify out just the `GridCanvas` component that just draws the data and emits the raw events. Then at a higher level, you can add a mouse handler that and combine it for the `headerEvent` (or other such events); and with floating rows/columns, they'd be separate `GridCanvas` (going back to the first point).